src/memory.rs

```diff
@@ -1,48 +1,109 @@
+use candid::Principal;
 use ic_stable_structures::memory_manager::{MemoryId, MemoryManager, VirtualMemory};
 #[cfg(target_arch = "wasm32")]
 use ic_stable_structures::DefaultMemoryImpl;
+use ic_stable_structures::StableBTreeMap;
 #[cfg(not(target_arch = "wasm32"))]
 use ic_stable_structures::VectorMemory;
-use ic_stable_structures::{Cell, StableBTreeMap};
 use std::cell::RefCell;
 
-use crate::{
-    constants::NODES_IN_FIDUCIARY_SUBNET,
-    types::{AuthSet, Metadata, Metrics, PrincipalStorable, Provider, StorableRpcService},
-};
+use crate::types::{ApiKey, Metrics, PrincipalStorable, ProviderId};
 
 #[cfg(not(target_arch = "wasm32"))]
 type Memory = VirtualMemory<VectorMemory>;
 #[cfg(target_arch = "wasm32")]
 type Memory = VirtualMemory<DefaultMemoryImpl>;
 
 thread_local! {
-    // Unstable static data: this is reset when the canister is upgraded.
+    // Unstable static data: these are reset when the canister is upgraded.
     pub static UNSTABLE_METRICS: RefCell<Metrics> = RefCell::new(Metrics::default());
-    static UNSTABLE_SUBNET_SIZE: RefCell<u32> = RefCell::new(NODES_IN_FIDUCIARY_SUBNET);
+    static UNSTABLE_DEMO_STATUS: RefCell<bool> = RefCell::new(false);
```

Comment:
I don't have a strong preference either way, but we might as well keep this in stable memory since that seems more intuitive for how people might use the canister in local development.

It also turns out `Storable` isn't implemented for `bool`, at least in the version of `ic-stable-structures` used in the project. 🙃 I'll add a `BoolStorable` for now.

Reply:
That's a good point. This is now implemented, but unfortunately `ic-stable-structures` (or at least the version in the canister) doesn't implement `Storable` for `bool`, so I had to include a `BoolStorable`. I'll include some tests for this as well as `StringStorable` and `PrincipalStorable` in a follow-up PR.